src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -113,6 +122,21 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
+            if (this.ParameterSetName.Equals(ArmParameterSet, StringComparison.Ordinal))
+            {
+                MetadataResponse metadataEndpoints = null;
+                metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
```

Comment:
We need to check the returned value is not null before dereferencing it (and throw a meaningful error if it is)